sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -176,630 +123,292 @@ public virtual Pageable<string> GetMetricDimensionValues(string metricId, string
             }
-
-            return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, null);
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
         }
 
         /// <summary>
-        /// Gets a collection of items describing the time series of a specified <see cref="DataFeedMetric"/> and
-        /// details about their ingested data points.
+        /// Adds a <see cref="MetricFeedback"/>.
         /// </summary>
-        /// <param name="metricId">The unique identifier of the <see cref="DataFeedMetric"/>.</param>
-        /// <param name="options">The set of options used to configure the request's behavior.</param>
+        /// <param name="feedback">The <see cref="MetricFeedback"/> to be created.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>A <see cref="Pageable{T}"/> containing the collection of <see cref="MetricSeriesData"/>.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="metricId"/> or <paramref name="options"/> is null.</exception>
-        /// <exception cref="ArgumentException"><paramref name="metricId"/> is empty or not a valid GUID.</exception>
-        public virtual Pageable<MetricSeriesData> GetMetricSeriesData(string metricId, GetMetricSeriesDataOptions options, CancellationToken cancellationToken = default)
+        /// <returns>
+        /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <see cref="MetricFeedback"/>
+        /// containing information about the newly added feedback.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="feedback"/> is null.</exception>
+        public virtual Response<MetricFeedback> AddFeedback(MetricFeedback feedback, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNull(options, nameof(options));
+            Argument.AssertNotNull(feedback, nameof(feedback));
 
-            Guid metricGuid = ClientCommon.ValidateGuid(metricId, nameof(metricId));
-            IEnumerable<IDictionary<string, string>> series = options.SeriesKeys.Select(key => key.Dimension);
-            MetricDataQueryOptions queryOptions = new MetricDataQueryOptions(ClientCommon.NormalizeDateTimeOffset(options.StartsOn), ClientCommon.NormalizeDateTimeOffset(options.EndsOn), series);
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(AddFeedback)}");
+            scope.Start();
 
-            Page<MetricSeriesData> FirstPageFunc(int? pageSizeHint)
+            try
             {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesData)}");
-                scope.Start();
+                RequestContent content = MetricFeedback.ToRequestContent(feedback);
```

Comment:
In a DPG, we wouldn't be validating feedback!= null, so we would probably want to check for null here.

Reply:
We already have a null check [here](https://github.com/Azure/azure-sdk-for-net/blob/f23229d00e7dfe975d1074cd3fed5594bd50d36d/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs#L317).